### PR TITLE
Update payment failure scenarios

### PR DESCRIPTION
**Pull request by ctrudeau-stripe**
r @adreyfus-stripe 

Fixing issue #15 

Created a bug when I added this. Removing it and going to add a new section on handling the various error methods found here https://stripe.com/docs/api/errors/handling in a future version. 

Tested: 
- Used 4242 test card & 3DS test card.

### CODE REVIEW

File: client/script.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ function handleSubscription(subscription) {
   if (payment_intent) {
     const { client_secret, status } = payment_intent;
 
-    if (status === 'requires_action' || status === 'requires_payment_method') {
+    if (status === 'requires_action') {
```
Comment (thorsten-stripe) [inline client/script.js:97]: The else block (https://github.com/stripe-samples/set-up-subscriptions/pull/20/files#diff-484245f14c67ce4fc2d956536e096dc0R110) now handles also the payment error case, right? So we should probably update the comment on line 110-111 to reflect that.

**Review by thorsten-stripe — APPROVED** [review]
Just the one nit regarding the comments in the else block, otherwise LGTM

**Comment by thorsten-stripe** [comment]
Testing here https://codesandbox.io/s/github/stripe-samples/set-up-subscriptions/tree/update-payment-failure-scenarios with card `4000000000009995` and the frontend spins endlessly because we don't catch the subscription creation error:
image

**Comment by ctrudeau-stripe** [comment]
We are going to add failure scenarios in updates, not with this update.

**Pull request merged**